asapdiscovery-data/asapdiscovery/data/backend/openeye.py

```diff
@@ -800,15 +821,19 @@ def set_SD_data(mol: oechem.OEMol, data: dict[str, str]) -> oechem.OEMol:
         OpenEye OEMol with SD data set
     """
     for key, value in data.items():
-        oechem.OESetSDData(mol, key, value)
+        if not len(value) == mol.NumConfs():
+            raise ValueError(
+                f"Length of data for tag '{key}' does not match number of conformers ({mol.NumConfs()}). "
+                f"Expected {mol.NumConfs()} but got {len(value)} elements."
+            )
+        for i, conf in enumerate(mol.GetConfs()):
+            oechem.OESetSDData(conf, key, value[i])
     return mol
 
 
-def _set_SD_data_repr(mol: oechem.OEMol, data: dict[str, str]) -> oechem.OEMol:
+def set_SD_data(mol: oechem.OEMol, data: dict[str, str]) -> oechem.OEMol:
```

Comment:
What form does the dict have to be in to sucessfully write multiconformer data? Worth explaining in docstring.

Reply:
If you want to set multiconf data, ie. set the POSIT probability or something for all the conformers, then you should use the `set_multiconf_SD_data` function with a `dict[str, list]`

I could change it so that if you pass `dict[str, str]` than it does that this function does and `dict[str, list]` then it does what the other function does. would that make more sense?

Reply:
Or I could make this (and the `get` methods) always use a `dist[str, list]` and for setting the data, if the list has length 1 then it sets it to all the confs, and if the length of the list matches it does what the multiconf method does (and if the lengths dont match it throws an error).